server/coordinator/procedure/operation/split/split.go

```diff
@@ -113,31 +144,43 @@ func (p *Procedure) Typ() procedure.Typ {
 	return procedure.Split
 }
 
+func (p *Procedure) RelatedVersionInfo() procedure.RelatedVersionInfo {
+	shardWithVersion := make(map[storage.ShardID]uint64, 0)
+	for _, shardView := range p.snapshot.Topology.ShardViews {
+		if shardView.ShardID == p.shardID {
+			shardWithVersion[p.shardID] = shardView.Version
+		}
+	}
+	shardWithVersion[p.newShardID] = 0
+	return procedure.RelatedVersionInfo{
+		ClusterID:        p.snapshot.Topology.ClusterView.ClusterID,
+		ShardWithVersion: shardWithVersion,
+		ClusterVersion:   p.snapshot.Topology.ClusterView.Version,
+	}
+}
+
+func (p *Procedure) Priority() procedure.Priority {
+	return procedure.PriorityHigh
+}
+
 func (p *Procedure) Start(ctx context.Context) error {
 	p.updateStateWithLock(procedure.StateRunning)
 
 	splitCallbackRequest := callbackRequest{
-		ctx:            ctx,
-		cluster:        p.cluster,
-		dispatch:       p.dispatch,
-		shardID:        p.shardID,
-		newShardID:     p.newShardID,
-		schemaName:     p.schemaName,
-		tableNames:     p.tableNames,
-		targetNodeName: p.targetNodeName,
+		ctx:             ctx,
+		snapshot:        p.snapshot,
+		dispatch:        p.dispatch,
+		shardID:         p.shardID,
+		newShardID:      p.newShardID,
+		schemaName:      p.schemaName,
+		tableNames:      p.tableNames,
+		targetNodeName:  p.targetNodeName,
+		clusterMetadata: p.clusterMetadata,
```

Comment:
The `callbackRequest` is private, and I guess we can just make p as a field of it.